### PR TITLE
e2e: host-level: verify NRT updates

### PR DESCRIPTION
**Pull request by shajmakh**
Add a check to verify NRT is updated accordingly for gu pods or not
updated for non-gu pods.

### CODE REVIEW

**Comment by shajmakh** [comment]
/hold
pending updates to [unsched] cases

**Comment by shajmakh** [comment]
/cc @ffromani 
just fyi for now, this is still pending more verification

File: internal/resourcelist/resourcelist.go [hunk 1]
```diff
@@ -119,6 +120,28 @@ func Accumulate(ress []corev1.ResourceList) corev1.ResourceList {
...
+				}
+			}
+
+			qty := ret[resName]
+			qty.Add(resQty)
+			ret[resName] = qty
+		}
+	}
+	return ret
+}
```
[the remark this answers is not in the document]
Comment (shajmakh) [reply internal/resourcelist/resourcelist.go:143]: This is a really nice idea, as it enables flexibility in resources accounting, yet I'd be in favor of it more when the filter function implementation is not too diverse and conditional. when we adopt it here, it seems to me that we are carrying half of the implementation in the filter function. I've added it anyway (in next update), let me know what you think.

File: test/e2e/serial/tests/resource_hostlevel.go [hunk 2]
```diff
@@ -108,6 +113,31 @@ var _ = Describe("[serial][hostlevel] numaresources host-level resources", Seria
...
+				e2efixture.MustSettleNRT(fxt)
+
+				targetNrtInitial, err := e2enrt.FindFromList(nrtList.Items, updatedPod.Spec.NodeName)
+				Expect(err).NotTo(HaveOccurred())
+
+				if expectedQOS == corev1.PodQOSGuaranteed {
+					By("verify resources consumption is reflected in the NRTs")
+					exclusiveRequired := intreslist.AccumulateExclusiveForNRT(requiredRes)
+					klog.Infof("exclusive required resources: %+v", exclusiveRequired)
+					expectNRTConsumedResources(fxt, *targetNrtInitial, exclusiveRequired, updatedPod)
```
[the remark this answers is not in the document]
Comment (shajmakh) [reply test/e2e/serial/tests/resource_hostlevel.go:127]: I tried to prefer clarity over brevity here, I'll fix it

**Comment by shajmakh** [comment]
testing on both TM scopes is in progress

**Comment by shajmakh** [comment]
passed on both scopes:

`
Ran 7 of 104 Specs in 423.545 seconds
SUCCESS! -- 7 Passed | 0 Failed | 0 Pending | 97 Skipped
`

File: test/e2e/serial/tests/resource_hostlevel.go [hunk 3]
```diff
@@ -108,6 +108,32 @@ var _ = Describe("[serial][hostlevel] numaresources host-level resources", Seria
...
+						hostLevel := resName == corev1.ResourceEphemeralStorage
+						excCpus := true
+						if resName == corev1.ResourceCPU {
+							_, err := strconv.Atoi(resQty.String())
+							if err != nil && resQty.Value()%1000 > 0 {
+								excCpus = true
+							}
+						}
+						return !(hostLevel || excCpus)
+
```
Comment (ffromani) [inline test/e2e/serial/tests/resource_hostlevel.go:130]: IIUC the intent:
```
if resName == corev1.ResourceEphemeralStorage {
    return false
}
if resName == corev1.ResourceCPU {
    cpuAmount, ok := cpuQty.AsInt64()
    if !ok {
        return false
    }
    if cpuAmount % 1000 > 0 {
        return false
    }
}
return true
```

**Review by ffromani — COMMENTED** [review]
not completely sure about the intent of a code snippet, otherwise LGTM

File: test/e2e/serial/tests/resource_hostlevel.go (hunk 3 above)
Comment (ffromani) [reply test/e2e/serial/tests/resource_hostlevel.go:130]: Feel free to borrow code (and point out mistakes, if any) from https://github.com/kubernetes-sigs/scheduler-plugins/blob/master/pkg/noderesourcetopology/resourcerequests/exclusive.go

Comment (shajmakh) [reply test/e2e/serial/tests/resource_hostlevel.go:130]: yes that's the intention, I'll fix the confusion

**Pull request merged**